nncf/quantization/waveq_loss.py

```diff
@@ -0,0 +1,59 @@
+import math
+import torch
+
+from nncf.compression_method_api import CompressionLoss
+
+
+class WaveQLoss(CompressionLoss):
+
+    def __init__(self, quantize_modules):
+        self.quantize_modules = quantize_modules
+        self.hook_handlers = None
+
+    def set_up_hooks(self):
```

Comment:
After destruction of WaveQ loss all hooks still will be called. I think it makes sense to delete them in
`def __del__(self):`